src/main/kotlin/org/rust/ide/editor/RustSmartEnterProcessor.kt

```diff
@@ -0,0 +1,39 @@
+package org.rust.ide.editor
+
+import com.intellij.lang.SmartEnterProcessorWithFixers
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+
+class RustSmartEnterProcessor : SmartEnterProcessorWithFixers() {
+    init {
+        addFixers(StatementFixer)
+        addEnterProcessors(PlainEnterProcessor)
+    }
+
+    private object StatementFixer : Fixer<RustSmartEnterProcessor>() {
+        override fun apply(editor: Editor, processor: RustSmartEnterProcessor, element: PsiElement) {
+            val doc = editor.document
+            val offset = element.textOffset
+            val lineNum = doc.getLineNumber(offset)
+            val lineStart = doc.getLineStartOffset(lineNum)
+            val lineEnd = doc.getLineEndOffset(lineNum)
+            val range = lineStart..lineEnd
+            val openParenthesisCount = range.count { doc.getText(TextRange(it, it + 1)) == "(" }
+            val closeParenthesisCount = range.count { doc.getText(TextRange(it, it + 1)) == ")" }
+            val num = openParenthesisCount - closeParenthesisCount
```

Comment:
I don't know exactly how to implement such fixers, but again, this should be based on the PSI structure, and not on the text, because, for example `let x = "(((".to_string()`. You can use `Find usages` on [`SmartEnterProcessor`](https://upsource.jetbrains.com/idea-ce/file/idea-ce-1731d054af4ca27aa827c03929e27eeb0e6a8366/java/java-impl/src/com/intellij/codeInsight/editorActions/smartEnter/JavaSmartEnterProcessor.java?nav=1915:1935:focused&line=0) to find some inheritors in IDEA.